src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,152 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+/**
+ * TaskList manages the usage of Task.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+    private Storage storage;
+
+    /**
+     * Takes in a Storage to save or load data.
+     * Returns an instance of TaskList.
+     */
+    public TaskList(Storage storage) {
+        this.tasks = new ArrayList<>();
+        this.storage = storage;
+    }
+
+    /**
+     * Loads data from storage.
+     */
+    public void load() throws DukeException {
+        try {
+            storage.load(this);
+        } catch (FileNotFoundException e) {
+            throw new DukeException.DukeIoException();
+        }
+    }
+
+    /**
+     * Saves data to storage.
+     */
+    public void save() throws DukeException {
+        try {
+            this.storage.save(tasks);
+        } catch (IOException e) {
+            throw new DukeException.DukeIoException();
+        }
+    }
+
+    /**
+     * Create task based on task's type.
+     *
+     * @param description of the task.
+     * @param time        of the task for Deadline and Event.
+     * @param t           type of task.
```

Comment:
Is the whitespace before each parameter description here is intended?

Reply:
Yes, it is formatted by intelliJ. Are the white space not supposed to be there?